### PR TITLE
Move optionparser to thirdparty [12336]

### PR DESCRIPTION
**Pull request by jparisu**
Tested with CMake Options:
```
-DCMAKE_BUILD_TYPE=Release
-DCOMPILE_EXAMPLES=ON
-DGTEST_INDIVIDUAL=ON
-DFASTRTPS_API_TESTS=ON
-DFASTDDS_PIM_API_TESTS=ON
-DPERFORMANCE_TESTS=ON
-DEPROSIMA_BUILD_TESTS=ON
-DVIDEO_TESTS=ON
```

Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: thirdparty/optionparser/optionparser.h [hunk 1]
```diff
@@ -1556,9 +1556,9 @@ inline bool Parser::workhorse(bool gnu, const Descriptor usage[], int numargs, c
 
     do // loop over short options in group, for long options the body is executed only once
     {
-      int idx(0);
+      int idx;
 
-      const char* optarg = nullptr;
+      const char* optarg;
```
Comment (JLBuenoLopez-eProsima) [inline thirdparty/optionparser/optionparser.h:1561]: Windows CI is complaining about this variables being uninitialized. I think we should revert these two changes.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
LGTM though Windows CI is issuing a lot of warnings that should be fixed before considering merging.

Also, fix please the uncrustify in those files that are not external.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with greenish CI

**Comment by jparisu** [comment]
Uncrustify fails in thirdparty file `optionparser.h`

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM, although I am not sure whether Linux tests are expected to fail. Also, perhaps we might want to preserve thirdparties in every example so they can be run as standalone programs.

**Pull request closed without merging**